Cubical/Core/Prelude.agda

```diff
@@ -47,10 +47,15 @@ cong f p = λ i → f (p i)
 
 -- This is called compPath and not trans in order to eliminate
 -- confusion with transp
-compPath : x ≡ y → y ≡ z → x ≡ z
-compPath {x = x} p q i =
-  hcomp (λ j → \ { (i = i0) → x
-                 ; (i = i1) → q j }) (p i)
+
+compPath-filler : ∀ {x y z : A} → x ≡ y → y ≡ z → I → I → A
+compPath-filler {x = x} p q j i =
+  hfill (λ j → λ { (i = i0) → x
+                  ; (i = i1) → q j }) (inc (p i)) j
+
+compPath : ∀ {x y z : A} → x ≡ y → y ≡ z → x ≡ z
+compPath {x = x} p q j =
```

Comment:
Just write `compPath p q = compPath-filler p q i1`

Reply:
@3abc that doesn't work because `compPath-filler p q i1` has a function type, but we are making a path.

Reply:
Oh OK.

Reply:
I suggest giving `compPath-filler` a more specific type. 

Maybe bring back the definition of `compPath` and write

```
compPath-filler : ∀ {x y z : A} → (p : x ≡ y) → (q : y ≡ z)
                → PathP (λ i → x ≡ q i) p (compPath p q)
compPath-filler {x = x} p q j i =
  hfill (λ j → λ { (i = i0) → x
                  ; (i = i1) → q j }) (inc (p i)) j
```

This helps to reduce user's mental overhead when they want to use `compPath-filler`

Reply:
> I suggest giving `compPath-filler` a more specific type.
> 
> Maybe bring back the definition of `compPath` and write
> 
> ```
> compPath-filler : ∀ {x y z : A} → (p : x ≡ y) → (q : y ≡ z)
>                 → PathP (λ i → x ≡ q i) p (compPath p q)
> compPath-filler {x = x} p q j i =
>   hfill (λ j → λ { (i = i0) → x
>                  ; (i = i1) → q j }) (inc (p i)) j
> ```
> This helps to reduce user's mental overhead when they want to use `compPath-filler`

I'd rather not have redundancy --- shouldn't the faces of the filler be explicitly clear from the hfill? It is natural to define the filler first and then extract the Kan composite from it, I think. That's one of the good practices of Redtt (see [this](https://github.com/RedPRL/redtt/blob/master/library/prelude/path.red) file). But that's just my opinion. :)

Reply:
> I'd rather not have redundancy --- shouldn't the faces of the filler be explicitly clear from the hfill?

One of the points of a library is that user can have an interface to use it without having to look at details of the implementation. Yes it is kinda clear from `hfill`. But for a user who want to use `compPath-filler` when doing something else, the type of `compPath-filler` is the user interface, and `hfill` is just how you implement `compPath-filler`, which is irrelevant to the user.

So writing it as `I → I → A` it's rather unenlightening.

Also, `hfill` will reduce to `hcomp` anyway so define `compPath` first is not redundant at all.

Reply:
Maybe at least put the more precise type `PathP (λ i → x ≡ q i) p (compPath p q)` in comment?

Reply:
> Maybe at least put the more precise type `PathP (λ i → x ≡ q i) p (compPath p q)` in comment?

That's a great idea! :+1:

Reply:
@mortberg What do you think? Should fillers come before their composite or the other way around? (I need to define more fillers for the heterogeneous groupoid laws.)

Reply:
Let's do fillers first, but with the type in a comment